public/dorequest.php

```diff
@@ -213,7 +213,10 @@ function DoRequestError($errorMsg): void
          * Prefer later values, i.e. allow AddEarnedAchievementJSON to overwrite the 'success' key
          */
         $response = array_merge($response, unlockAchievement($user, $achIDToAward, $hardcore));
-        $response['Score'] = getPlayerPoints($user);
+        $response['Score'] = 0;
+        if (getPlayerPoints($user, $userPoints)) {
+            $response['Score'] = $userPoints['RAPoints'] + $userPoints['RASoftcorePoints'];
+        }
```

Comment:
This _definitely_ should only be returning the hardcore points. It's used in the client:
![image](https://user-images.githubusercontent.com/32680403/174445414-3d6cc508-812f-4eae-a937-0e9e3bf868cf.png)

Which I now realize will likely need supporting changes as well. You'll need to pass `SoftcorePoints` out too. I don't think we need a copy of `Points` in `HardcorePoints`.

Reply:
Also, whatever you do here will need to be done in `authenticateFromPasswordOrAppToken` so the initial values are known. This just updates the values after an achievement is unlocked.

Reply:
Thanks for the insight on this part as well as the API files as well. These were ones I had some issues with and I may need some more assistance with this as I'm seeing issues in RALib as I test earning achievements after I earn one the Total Points displayed go to 0 until I reload RALib so I likely still have something wrong between this or the API. Not sure if there is a good way to get more details from RALib (do I need to build the debug version?) or if this is related to the changes that you mention will need to be done on that end

Reply:
It's a data typing issue.
Prod:
```
{
 "Success":false,"AchievementsRemaining":12,
 "Error":"User already has this achievement unlocked in hardcore mode.",
 "Score":152872,"AchievementID":3159
}
```
Test:
```
{
 "Success":false,"AchievementsRemaining":18,
 "Error":"User already has this achievement unlocked in hardcore mode.",
 "Score":"75650","SoftcoreScore":"1100","AchievementID":3159
}
```
Make sure to cast `Score` and `SoftcoreScore` to integers.

Reply:
That was it thanks again! I believe the overall issue was when I changed `getPlayerPoints` I didn't preserve the typing when changing it to instead pass `$dataOut`